Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.cpp

```diff
@@ -75,24 +72,53 @@ void CommonAsmRoutines::GenFrsqrte()
   MOVQ_xmm(XMM0, R(RSCRATCH2));
   RET();
 
-  // Exception flags for zero input.
-  SetJumpTarget(zero);
+  SetJumpTarget(complex);
+  AND(32, R(RSCRATCH_EXTRA), Imm32(0x7FF));
+  CMP(32, R(RSCRATCH_EXTRA), Imm32(0x7FF));
+  FixupBranch nan_or_inf = J_CC(CC_E);
+
+  MOV(64, R(RSCRATCH2), R(RSCRATCH));
+  SHL(64, R(RSCRATCH2), Imm8(1));
+  FixupBranch nonzero = J_CC(CC_NZ);
+
+  // +0.0 or -0.0
   TEST(32, PPCSTATE(fpscr), Imm32(FPSCR_ZX));
-  FixupBranch skip_set_fx1 = J_CC(CC_NZ);
+  FixupBranch skip_set_fx3 = J_CC(CC_NZ);
   OR(32, PPCSTATE(fpscr), Imm32(FPSCR_FX | FPSCR_ZX));
-  FixupBranch complex3 = J();
+  SetJumpTarget(skip_set_fx3);
+  MOV(64, R(RSCRATCH2), Imm64(0x7FF0'0000'0000'0000));
+  OR(64, R(RSCRATCH2), R(RSCRATCH));
+  MOVQ_xmm(XMM0, R(RSCRATCH2));
+  RET();
+
+  // SNaN or QNaN or +Inf or -Inf
+  SetJumpTarget(nan_or_inf);
+  MOV(64, R(RSCRATCH2), R(RSCRATCH));
+  SHL(64, R(RSCRATCH2), Imm8(12));
+  FixupBranch inf = J_CC(CC_Z);
+  BTS(64, R(RSCRATCH), Imm8(51));  // Silence NaN
```

Comment:
Any reason this comment is on the same line instead of before? The others mark the beginning of a block, and this one seems no different to me.